dns.go

```diff
@@ -45,10 +46,34 @@ type DNSListResponse struct {
 	ResultInfo `json:"result_info"`
 }
 
+// nontransitionalLookup implements the nontransitional processing as specified in
+// Unicode Technical Standard 46 with almost all checkings off to maximize user freedom.
+var nontransitionalLookup = idna.New(
+	idna.MapForLookup(),
+	idna.StrictDomainName(false),
+	idna.ValidateLabels(false),
+)
+
+// normalizeDomainName tries to convert IDNs (international domain names)
+// from Unicode form to Punycode, using non-transitional process specified
+// in UTS 46.
+//
+// Confusinigly, this is in the opposite direction of normalizeZoneName
+// to work around the current quirks of the Cloudflare servers.
+//
+// Note: conversion errors are silently discarded and partial conversion
+// results are used.
+func normalizeDomainName(name string) string {
```

Comment:
this seems like a poorly named method (`normalizeZoneName` isn't any better IMO). can we instead rename this to cover what the method does? i.e. `stringToUTS46ASCII()` or similar? ideally this would it's own type that we could pass a string into and perform the comparison independent of the DNS structs.

Reply:
i.e. implement the [stringer interface](https://pkg.go.dev/fmt#Stringer) for these types

Reply:
> this seems like a poorly named method (`normalizeZoneName` isn't any better IMO). can we instead rename this to cover what the method does? i.e. `stringToUTS46ASCII()` or similar? ideally this would it's own type that we could pass a string into and perform the comparison independent of the DNS structs.

Will change the name. Ideally, we would not need any of these functions once the API is fixed. Should we rename or even remove `normalizeZoneName` (in a separate PR)? As far as I know, it's not needed anymore.

Reply:
> i.e. implement the [stringer interface](https://pkg.go.dev/fmt#Stringer) for these types -- https://play.golang.org/p/vp00zNkQkMS

I see. How about something like this?
```go
type DNSRecordName string
```
More details: https://play.golang.org/p/GmhyEQwEFAf The benefit is that it can then be used in all different data structures.

Reply:
> I see. How about something like this?
> 
> ```go
> type DNSRecordName string
> ```
> 
> More details: https://play.golang.org/p/GmhyEQwEFAf The benefit is that it can then be used in all different data structures.

On the second thought, maybe I should keep `string` as `string` so that the API remains the same.

Reply:
So I changed `normalizeDomainName` to `toUTS46ASCII`, hopefully addressing the issue. I did not implement the `toASCII` method for those structures because I am not sure for what structures I should implement it. Also, when other domain-like data are present (e.g., emails), I am not sure if the method should apply the conversion to them as well.